raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3426,6 +3426,12 @@ public Optional<SnapshotWriter<T>> createSnapshot(
             throw new IllegalStateException("Cannot create snapshot before the replica has been initialized");
         }
 
+        long baseOffset = log.read(snapshotId.offset(), Isolation.COMMITTED).startOffsetMetadata.offset();
```

Comment:
Is there a reason why you added this check here and not to `ReplicatedLog` (`MockLog` and `KafkaMetadataLog`)? Those types already do snapshot id validation against the underlying log records.

Reply:
Looking at this again, it makes more sense to have this moved to the `ReplicatedLog` level.